docs/reference/modules/scripting/groovy.asciidoc

```diff
@@ -0,0 +1,144 @@
+[[modules-scripting-groovy]]
+== Groovy Scripting Language
+
+.Groovy dynamic scripting off by default from v1.4.3
+[IMPORTANT]
+===================================================
+
+Groovy dynamic scripting is off by default. This prevents Groovy scripts
+from being accepted as part of a request or retrieved from the
+`.scripts` index. You can still use Groovy file scripts stored in
+the `config/scripts/` directory on every node.
+
+To convert an inline script to a file-based script, save the contents
+of the `inline` field to a file with the `.groovy` extension and
+store it in the `config/scripts` directory on every data node in your
+cluster.
+
+For example, if you have the following inline script:
+
+[source,js]
+-----------------------------------
+GET /_search
+{
+    "script_fields": {
+        "my_field": {
+            "inline": "1 + my_var",
+            "params": {
+              "my_var": 2
+            }
+        }
+    }
+}
+-----------------------------------
+
+Save `1 + my_var` in a file called `config/scripts/my_script.groovy`.
+
+To use the script in a request, specify its name (without the `.groovy` extension) in the `file` field:
+
+[source,js]
+-----------------------------------
+GET /_search
+{
+    "script_fields": {
+        "my_field": {
+            "script": {
+                "file": "my_script",
+                "params": {
+                    "my_var": 2
+                }
+            }
+        }
+    }
+}
+-----------------------------------
+
+===================================================
+
+[float]
+=== Groovy Built In Functions
```

Comment:
I'm a bit confused as to why this is provided as documentation?  Is this something we do outside of Groovy or could this also be found in the Groovy documentation?  (I realize this is reformatted so it was here before, but I was curious.)